### PR TITLE
Added movie info pop up

### PR DESCRIPTION
**Pull request by younessoudouh**
-created modal component
-created movie-info component
-created custom hook for movie info API and movie trailer API
Screenshot (15)

### CODE REVIEW

File: src/Componnets/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.modal {
+    position: fixed;
+    height: 100vh;
+    width: 100vw;
+    background-color: #22242399;
+    z-index: 999;
```
Comment (faouzioudouh) [inline src/Componnets/modal/modal.css:6]: why `999` ?

File: src/Componnets/modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.modal {
+    position: fixed;
+    height: 100vh;
+    width: 100vw;
+    background-color: #22242399;
```
Comment (faouzioudouh) [inline src/Componnets/modal/modal.css:5]: Wrong color?

File: src/Componnets/movie-info/movie-info.css [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+.watch-trailer {
+    color: #fff;
+    background-color: #f50057;
+    text-decoration: none;
+    text-transform: uppercase;
+    padding: 6px 16px;
+    font-size: 16px;
+    transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+    font-family: "Roboto", "Helvetica", "Arial", sans-serif;
```
Comment (faouzioudouh) [inline src/Componnets/movie-info/movie-info.css:58]: Is this a copy past from the website?

File: src/Componnets/movie-info/movie-info.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  const { data: movieTrailer } = useMovieTrailerApi(mediaType, id);
+
+  const YOUTUBE_KEY = movieTrailer && movieTrailer[0]?.key;
+
+  const POSTER__URL =
+    (movieInfo?.poster_path &&
+      "https://image.tmdb.org/t/p/w300" + movieInfo.poster_path) ||
+    "https://www.movienewz.com/img/films/poster-holder.jpg";
+
+  const handlePropagation = (event) => {
```
Comment (faouzioudouh) [inline src/Componnets/movie-info/movie-info.jsx:19]: Why do you need this?

File: src/api/api-movie-trailer.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+import { useQuery } from "react-query";
+
+export const useMovieTrailerApi = (mediaType, id) => {
```
Comment (faouzioudouh) [inline src/api/api-movie-trailer.js:4]: `useMovieTrailerApi` => `useMovieTrailer`

File: src/Componnets/modal/modal.css (hunk 1 above)
Comment (younessoudouh) [reply src/Componnets/modal/modal.css:6]: we want it to stay on the top of everthing

File: src/Componnets/movie-info/movie-info.css (hunk 3 above)
Comment (younessoudouh) [reply src/Componnets/movie-info/movie-info.css:58]: no

File: src/Componnets/movie-info/movie-info.jsx (hunk 4 above)
Comment (younessoudouh) [reply src/Componnets/movie-info/movie-info.jsx:19]: to show movie info in pop up we should fetch movie info and trailer as well

Comment (younessoudouh) [reply src/Componnets/movie-info/movie-info.jsx:19]: we have on click action in movie info parent and in movie info as well so to stop bubbling when click on movie info

File: src/Componnets/modal/modal.css (hunk 2 above)
Comment (younessoudouh) [reply src/Componnets/modal/modal.css:5]: how?

File: src/Componnets/modal/modal.css (hunk 1 above)
Comment (faouzioudouh) [reply src/Componnets/modal/modal.css:6]: Can you use a reasonable number? like 3 or 4..

File: src/Componnets/movie-info/movie-info.css (hunk 3 above)
Comment (faouzioudouh) [reply src/Componnets/movie-info/movie-info.css:58]: Why do you need `font-family`?

File: src/Componnets/modal/modal.css [hunk 6]
```diff
@@ -2,7 +2,7 @@
     position: fixed;
     height: 100vh;
     width: 100vw;
-    background-color: #22242399;
+    background-color: #00000080;
```
Comment (faouzioudouh) [inline src/Componnets/modal/modal.css:5]: I still see this wrong color code?

**Review by faouzioudouh — APPROVED** [review]

**Pull request merged**